src/components/Layout/Navbar/components/UserProfileWithGitHubLogin.tsx

```diff
@@ -0,0 +1,73 @@
+import { Box, Button, Text } from "@chakra-ui/react";
+import Image from "next/image";
+import Link from "next/link";
+import { Dispatch, SetStateAction } from "react";
+
+import { LINKS } from "../../../../constants/AppConstants";
+import dummyImage from "../../../../../public/images/Avatar.png";
+import downArrow from "../../../../../public/icons/icons8-arrow.png";
+
+import styles from "./userProfile.module.css";
+
+export function UserProfileWithGitHubLogin({
+  isLoggedIn,
+  isDropdownVisible,
+  firstName,
+  imageURL,
+  setIsDropdownVisible,
+}: {
+  isLoggedIn: boolean;
+  isDropdownVisible: boolean;
+  firstName: string | null;
+  imageURL: string | null;
+  setIsDropdownVisible: Dispatch<SetStateAction<boolean>>;
+}) {
+  const altText: string = firstName ?? "user image";
+  const imageToShow = imageURL || dummyImage;
+
+  return (
+    <>
+      {isLoggedIn ? (
+        <Box
+          data-testid="userProfile"
+          className={styles.userprofile_container}
+          onClick={() => setIsDropdownVisible(!isDropdownVisible)}
+        >
+          <Text
+            className={styles.userprofile_user__first_name}
+          >{`Hello, ${firstName}`}</Text>
+          <Box className={styles.userprofile_image__wrapper}>
+            <Image
+              src={imageToShow}
+              style={{ borderRadius: "50%" }}
+              width={32}
+              height={32}
+              alt={altText}
+            />
+            <Image width={15} height={15} src={downArrow} alt="downArrow" />
```

Comment:
Since this is a image, probably good idea to add alt text.

Reply:
already added alt text

Reply:
I meant alt text for the icon image.

Reply:
Added alt text for icon downArrow